src/scripts/main.js

```diff
@@ -1,3 +1,202 @@
+    const bNum = parseSalary(bText);
+
+    if (!isNaN(aNum) && !isNaN(bNum)) {
+      if (ascSort) {
+        return bNum - aNum;
+      }
+
+      return aNum - bNum;
+    } else {
+      if (ascSort) {
+        return bText.localeCompare(aText);
+      }
+
+      return aText.localeCompare(bText);
+    }
+  });
+
+  sortedRows.forEach((row) => tBody.appendChild(row));
+}
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    if (row.classList.contains('active')) {
+      row.classList.remove('active');
+    } else {
+      rows.forEach((r) => r.classList.remove('active'));
+      row.classList.add('active');
+    }
+  });
+});
+
+const form = document.createElement('form');
+const button = document.createElement('button');
+
+const attributes = ['name', 'position', 'office', 'age', 'salary'];
+
+const options = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+for (let i = 0; i <= 4; i++) {
+  const input = document.createElement('input');
+  const label = document.createElement('label');
+  const select = document.createElement('select');
+
+  const attr = attributes[i];
+
+  label.textContent = `${attr.charAt(0).toUpperCase() + attr.slice(1, attr.length)}:`;
+
+  input.setAttribute('data-qa', attr);
+  input.setAttribute('required', 'true');
+  input.setAttribute('name', attr);
+  input.setAttribute('type', 'text');
+
+  if (attr === 'age' || attr === 'salary') {
+    input.setAttribute('type', 'number');
+  }
+
+  select.setAttribute('name', 'office');
+  select.setAttribute('data-qa', 'office');
+
+  label.appendChild(input);
+
+  if (attr === 'office') {
+    const labelSelect = document.createElement('label');
+
+    labelSelect.textContent = `${attr.charAt(0).toUpperCase() + attr.slice(1, attr.length)}:`;
+
+    for (let j = 0; j < options.length; j++) {
+      const option = document.createElement('option');
+
+      const optionText = options[j];
+
+      option.value = optionText;
+
+      const text =
+        optionText.charAt(0).toUpperCase() + optionText.slice(1, option.length);
+
+      option.textContent = text;
+
+      select.appendChild(option);
+      labelSelect.appendChild(select);
+    }
+
+    form.appendChild(labelSelect);
+  } else {
+    form.appendChild(label);
+  }
+}
+
+button.textContent = 'Save to table';
+
+form.appendChild(button);
+
+form.classList.add('new-employee-form');
+document.documentElement.appendChild(form);
+
+function formatSalary(salary) {
+  return `$${Number(salary).toLocaleString('en-US')}`;
+}
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const data = Object.fromEntries(formData.entries());
+
+  const tr = document.createElement('tr');
+
+  for (const key in data) {
+    const td = document.createElement('td');
+
+    if (+data[key] > 100) {
+      td.textContent = formatSalary(data[key]);
+    } else {
+      td.textContent = data[key];
+    }
```

Comment:
The logic for formatting salary values might not work as intended. The condition `if (+data[key] > 100)` assumes that any value over 100 is a salary, which might not always be the case. Consider checking the key or attribute name to ensure correct formatting.